supabase/migrations/20250523191212_assignment-eval-settings.sql

```diff
@@ -0,0 +1,245 @@
+    "allow_early" boolean,
+    "class_id" bigint not null 
+);
+
+ALTER TABLE "public"."self_review_settings" ENABLE ROW LEVEL SECURITY;
+
+CREATE POLICY "anyone in the course can view self review settings" 
+ON "public"."self_review_settings"
+AS PERMISSIVE
+FOR SELECT
+USING (
+    authorizeforclassinstructor(class_id) OR authorizeforclassgrader(class_id) 
+    OR authorizeforclass(class_id)
+);
+
+CREATE POLICY "instructors can insert self review settings" 
+ON "public"."self_review_settings"
+AS PERMISSIVE
+FOR INSERT
+WITH CHECK (
+    authorizeforclassinstructor(class_id)
+);
+
+CREATE POLICY "instructors can update self review settings" 
+ON "public"."self_review_settings"
+AS PERMISSIVE
+FOR UPDATE
+USING (
+    authorizeforclassinstructor(class_id) 
+);
+
+CREATE POLICY "instructors can delete review settings" 
+ON "public"."self_review_settings"
+AS PERMISSIVE
+FOR DELETE
+USING (
+    authorizeforclassinstructor(class_id)
+);
+
+ALTER TABLE ONLY "public"."self_review_settings"
+    ADD CONSTRAINT "self_review_settings_pkey" PRIMARY KEY ("id");
+
+ALTER TABLE "public"."self_review_settings"
+    ADD CONSTRAINT "self_review_settings_class_fkey" FOREIGN KEY ("class_id") REFERENCES "public"."classes"("id");
+
+ALTER TABLE "public"."assignments" ADD COLUMN "self_review_setting_id" bigint NOT NULL;
+
+ALTER TABLE "public"."assignments"
+    ADD CONSTRAINT "assignments_self_review_setting_fkey" FOREIGN KEY ("self_review_setting_id") REFERENCES "public"."self_review_settings"("id");
+
+CREATE POLICY "Students can give themselves negative deadline exceptions"
+ON "public"."assignment_due_date_exceptions"
+AS PERMISSIVE 
+FOR INSERT
+WITH CHECK (
+   "hours" < 0 
+);
+
+CREATE POLICY "Deadline exceptions never inserted after early finish"
+ON "public"."assignment_due_date_exceptions"
+AS RESTRICTIVE 
+FOR INSERT
+WITH CHECK (
+   NOT EXISTS (
+    SELECT 1 FROM "public"."assignment_due_date_exceptions" adde
+    WHERE (student_id = adde.student_id OR assignment_group_id = adde.assignment_group_id)
+    AND assignment_id = adde.assignment_id AND class_id = adde.class_id AND adde.hours < 0
+   )
+);
+
+CREATE OR REPLACE FUNCTION public.assignments_grader_config_auto_populate()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+    declare 
+    rubric_id int;
+    self_rubric_id int;
+    begin
+  
+  INSERT INTO autograder (id, class_id) VALUES (NEW.id, NEW.class_id);
+  INSERT INTO autograder_regression_test (autograder_id,repository) VALUES (NEW.id, NEW.template_repo);
+  INSERT INTO rubrics (name, class_id, assignment_id) VALUES ('Grading Rubric', NEW.class_id, NEW.id) RETURNING id into rubric_id;
+  INSERT INTO rubrics (name, class_id, assignment_id) VALUES ('Self Rubric', NEW.class_id, NEW.id) RETURNING id into self_rubric_id;
+  UPDATE assignments set grading_rubric_id=rubric_id WHERE id=NEW.id;
+UPDATE assignments set self_review_rubric_id=self_rubric_id WHERE id=NEW.id;
+  RETURN NULL;
+end;$function$
+;
+
+-- TEMPORARY: REFINE!!!!!!!
```

Comment:
I think that the simplest security model for this RLS (and the auto_assign_self_reviews function) is:
- Only instructors can directly insert review_assignments
- auto_assign_self_reviews can not be invoked by anyone other than postgres, gets set as SECURITY DEFINER

This way, when the auto_assign_self_reviews function is invoked by a trigger (using logic that we control), it can create review_assignments.